include/bitcoin/bitcoin/chain/header.hpp

```diff
@@ -160,10 +160,11 @@ class BC_API header
     static uint256_t proof(uint32_t bits);
 
     bool is_valid_timestamp(uint32_t timestamp_limit_seconds) const;
-    bool is_valid_proof_of_work(uint32_t proof_of_work_limit) const;
-
-    code check(uint32_t timestamp_limit_seconds, uint32_t proof_of_work_limit)
+    bool is_valid_proof_of_work(uint32_t proof_of_work_limit, bool scrypt)
```

Comment:
Done. But I'd like to raise awareness that default arguments make the API more difficult to use: if someone accidentally drops the `scrypt` parameter during a refactoring then it will silently default to `false` and it'll take some time to track that down (or even notice) - especially if it's passed along a long call chain like here.